### DISCUSSION TITLE
How to show WSMan traffic to AMT?

### DISCUSSION THREAD

**Discussion by crass** [Ylianst/MeshCentral#4790] [General]
I'm trying to see all WSMan requests and responses to an AMT, either in MeshCentral or MeshCommander. I've looked at the [debugging information](https://ylianst.github.io/MeshCentral/meshcentral/debugging/) on the website for MeshCentral, and turned on all the logging channels as listed there (are there others not mentioned? notable there's not a `wsman` channel). Messages are logged to the log file, but no wsman traffic.

I've also running the webserver version of MeshCommander with the arguments `-wsmantrace -debug -logfile:/tmp/log.txt`. I see no `log.txt` being created, nor are there any messages in the developer console (using firefox). Do these options currently work?

How can I see the WSMan traffic without resorting to packet sniffing?

**Comment by jsastriawan** [comment]
Ok, first of all, wsmantrace flag applies to node webkit version.

If you want to trace Meshcentral WSMAN, look at webserver.js function handleRelayWebsocket. Have a look at the websocket data sent back and forth on that function. Alternatively, you can use browser debugging utility, look at websocket traffic at webrelay.ashx. You should see all WSMAN traffic there. You may not find the right HTTP authentication header in that traffic because it is being taken care of by interceptor.js at the server side.

The same tracing method for Meshcentral also applies meshcommander webserver edition.

If you want more detail, feel free to email me at joko dot sastriawan at intel.com. Additionally Open AMT Cloud Toolkit project team (sister project to this) may also able to help you.

**Reply by crass** [reply]
Thanks @jsastriawan for the tips. I'm unfamiliar with debugging websockets, using firefox, and didn't find a way to see the data in the network requests tab. Looking at it again, I see that for websocket connections both the request and response are in the response tab for the connection. Also, for MeshCommander, I am seeing the HTTP Authorization header there too.

I think it could be a nice feature to add a `wsman` logging target that logs all wsman requests and responses. Perhaps even have a separate log like the auth log.

Also, I appreciate the reference to the Open AMT Cloud Toolkit project there does indeed look like some useful tools there.
